### PR TITLE
feat(data/mmcif): Add default values for non-essential mmCIF values to handle incomplete mmCIF

### LINKED ISSUE #58: Support incomplete mmcif structure parsing
### Pre-flight checks

- [x] I searched existing issues and didn't find a match.
- [x] I can reproduce this on a clean environment (fresh venv).

### Summary

AF3 prediction result로 생성되는 cif에는 기존 실험 구조 mmCIF에 존재하는 항목 몇개 
- pdbx_audit_revision_history
- _exptl 
- _struct_keywords 
- pdbx_struct_assembly 
- pdbx_struct_oper_list 
- AtomSite.auth_comp_id
- Entity.pdbx_description
- ndb_seq_num
가 존재하지 않아 parsing이 안되는데 이러한 부분을 해결해서 AF3 prediction result cif를 gmol-base로 읽고자합니다. 

### Minimal reproducible example

```python
from pathlib import Path
from gmol.base.data.mmcif.parse import load_mmcif_single

# Any AF3 model CIF file
load_mmcif_single(Path("path/to/af3_model.cif"))
# → ValidationError: min() iterable argument is empty
```

### Expected behavior

AF3 predicted result 또한 없는 항목은 empty / default value를 가진채로 load 

### Actual behavior / logs / stack trace

```text
ValidationError: min() iterable argument is empty
```

### Environment

.

### Additional context / screenshots

_No response_

### PR DESCRIPTION
**Pull request by rladbtls1998**
## Checklist

- [x] Have you checked that there aren't other open [pull requests](http://github.com/seoklab/gmol-base/pulls) for the same issue?
- [x] Have you [linked the issue(s)](#linked-issues) you are working on (if any)?

If the change is related to the source code, tests, or build environments, please also check the following:

- [x] Does `pytest -vs` pass without any errors and warnings (at the project root)?
- [x] Does `mypy --pretty` pass without any errors and warnings (at the project root)?

If you added new feature(s), then also check the following:

- [x] Did you also add corresponding tests?

## Linked Issues

Link the tracking issue(s) of this PR, with the [auto-close keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) here:

- Closes #58

---

---

> [!NOTE]
> **Medium Risk**
> Changes mmCIF parsing defaults and validation behavior for several fields, which could subtly alter how incomplete/edge-case structures are interpreted and serialized. Scope is limited to metadata/assembly parsing and covered by new regression tests.
> 
> **Overview**
> Improves robustness to *incomplete mmCIF files* by adding defaults and fill-in logic for non-essential fields during parsing (e.g., empty `exptl_method`/`pdbx_keywords`, fallback revision date, optional `Entity.pdbx_description`, default `StructConn` values, and deriving missing `auth_comp_id`/nonpoly `seq_id`).
> 
> Updates assembly generation/serialization to handle missing metadata cleanly: `mmcif_assemblies` now returns the initial model with a warning when `pdbx_struct_assembly` is absent, and `Assembly.to_mmcif` writes `.` instead of emitting `None` for missing entity descriptions.
> 
> Adds a regression test plus a new fixture mmCIF (`metadata_incomplete_8aq1.cif`) to validate loading, assembly creation, and round-tripping without leaking `None` strings.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 036b1c1fbf1603260bc05a17c0e3b4c4ced79dfc. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/gmol/base/data/mmcif/parse.py [hunk 1]
```diff
@@ -142,6 +142,7 @@ class Scheme(LooseModel):
             "seq_id",  # poly_seq_scheme
             "ndb_seq_num",  # nonpoly_scheme
             "num",  # branch_scheme
+            "pdb_seq_num",  # fallback for nonpoly_scheme, branch_scheme, poly_seq_scheme
```
Comment (jnooree) [inline src/gmol/base/data/mmcif/parse.py]: [기능] 아.. 이게 딱 nonpoly scheme만 문제가 되네요. 제가 내일 좀 더 자세히 살펴보고 코멘트 추가로 드리겠습니다.

**Review by jnooree — CHANGES_REQUESTED** [review]

File: src/gmol/base/data/mmcif/parse.py [hunk 2]
```diff
@@ -439,8 +456,8 @@ class Mmcif(LooseModel):
 
     atom_site: list[AtomSite]
 
-    pdbx_struct_assembly: list[BioAssembly]
-    pdbx_struct_oper_list: dict[str, SymOp]
+    pdbx_struct_assembly: list[BioAssembly] = Field(default_factory=list)
+    pdbx_struct_oper_list: dict[str, SymOp] = Field(default_factory=dict)
```
[the remark this answers is not in the document]
Comment (rladbtls1998) [reply src/gmol/base/data/mmcif/parse.py]: 혹시 같은 dataclass의 기존에 존재하던 다른 mutable default들과 다르게 이 둘만 default_factory없이 mutable을 그대로 default value로 써야 하는 이유가 따로 있을까요? 이러면 한번에 여러 mmCIF를 gmol-base를 통해 처리할 때 오류가 생기지 않나요? 잘 몰라서 여쭤봅니다!

[the remark this answers is not in the document]
Comment (jnooree) [reply src/gmol/base/data/mmcif/parse.py]: 아 pydantic의 경우에는 mutable default 주어도 괜찮습니다!

https://docs.pydantic.dev/latest/concepts/fields/#mutable-default-values

File: src/gmol/base/data/mmcif/parse.py [hunk 3]
```diff
@@ -460,10 +477,10 @@ def _coerce_pdbx_kwds(v: Any) -> Any:
...
     @staticmethod
     def _find_oldest(v: dict[str, list[dict[str, Any]]]):
-        min_rev = min(
-            (rev for rev in v.get("pdbx_audit_revision_history", [])),
-            key=lambda r: int(r["ordinal"]),
-        )
+        revisions = v.get("pdbx_audit_revision_history", [])
+        if not revisions:
+            revisions = [{"ordinal": 1, "revision_date": "1970-01-01"}]
+        min_rev = min(revisions, key=lambda r: int(r["ordinal"]))
```
[the remark this answers is not in the document]
Comment (rladbtls1998) [reply src/gmol/base/data/mmcif/parse.py]: 반영했습니다!

**Review by jnooree — APPROVED** [review]
감사합니다!

File: src/gmol/base/data/mmcif/assembly.py [hunk 4]
```diff
@@ -139,7 +142,7 @@ class BranchPartner:
     atom_id: str
     leaving_atom_id: str
 
-    comp_id: str = field(compare=False)
+    comp_id: str
```
[the remark this answers is not in the document]
Comment (jnooree) [reply src/gmol/base/data/mmcif/assembly.py]: @rladbtls1998 의견 부탁드립니다!
